filip/models/ngsi_v2/base.py

```diff
@@ -202,19 +215,23 @@ class NamedMetadata(Metadata):
                     "ones: control characters, whitespace, &, ?, / and #.",
         max_length=256,
         min_length=1,
-        regex=FiwareRegex.standard.value  # Make it FIWARE-Safe
     )
-
-    @root_validator
-    def validate_data(cls, values):
-        if values.get("name", "").casefold() in ["unit",
-                                                 "unittext",
-                                                 "unitcode"]:
-            values.update(validate_unit_data(values))
-        return values
+    valid_name = field_validator("name")(validate_fiware_datatype_standard)
+
+    @model_validator(mode='after')
+    def validate_data(self):
+        if self.model_dump().get("name", "").casefold() in ["unit",
+                                                            "unittext",
+                                                            "unitcode"]:
+            valide_dict = self.model_dump()
```

Comment:
I think it should work, because model_validate should return a Unit object in this case. I implement a simple test `test_unit_as_metadata` in `test_base.py`. Maybe you can have double check there.